src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -3083,7 +3092,6 @@ void EmitLazy(RegexNode node)
 
                 // Iteration body
                 MarkLabel(body, emitSemicolon: false);
-                EmitTimeoutCheckIfNeeded();
```

Comment:
Wouldn't a Lazy loop also potentially backtrack and therefore require a timeout check?

Reply:
It can backtrack, but backtracking with a lazy loop is doing another iteration, which we check for and emit a timeout for later.